### PR TITLE
modify class MKLDNNActForward to new structure

### PR DESCRIPTION
**Pull request by rongzha1**
## Description ##
change class Act to new structure.

### Essentials ###
- [done] Passed code style checking (`make lint`)
- [done] Changes are complete (i.e. I finished coding on this PR)
- [done] All changes have test coverage
- [done] For user-facing API changes, API doc string has been updated. For new C++ functions in header files, their functionalities and arguments are well-documented. 
- [done] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [done] Feature1, tests, (and when applicable, API doc)
modify class MKLDNNActForward to new structure

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/nn/mkldnn/mkldnn_act.cc [hunk 1]
```diff
@@ -107,36 +107,46 @@ class MKLDNNActForward {
...
+};  // class MKLDNNActForward
 
-  void SetNewMem(const mkldnn::memory &data, const mkldnn::memory &output) {
-    if (this->data == nullptr)
+void MKLDNNActForward::SetDataHandle(const NDArray &in_data, const NDArray &out_data) {
+    auto input_mem = in_data.GetMKLDNNData();
+    auto out_mem = const_cast<NDArray &>(out_data).CreateMKLDNNData(
+        fwd_pd.dst_primitive_desc());
+
+    if (this->data == nullptr) {
```
Comment (TaoLv) [inline src/operator/nn/mkldnn/mkldnn_act.cc]: I think it would be better to add initialization and first creation in the constructive function the MKLDNNActForward class. So there is no need to chech nullptr for every iteration.
Besides, please follow the coding style and naming methods we discussed in previous email.

File: src/operator/nn/mkldnn/mkldnn_act.cc [hunk 2]
```diff
@@ -107,36 +107,46 @@ class MKLDNNActForward {
                    const NDArray &data, const mkldnn::memory &mem): fwd_pd(
```
Comment (TaoLv) [inline src/operator/nn/mkldnn/mkldnn_act.cc]: There is no need to input both data and mem. You can get mem from data easily.

File: src/operator/nn/mkldnn/mkldnn_act.cc [hunk 3]
```diff
@@ -107,36 +107,46 @@ class MKLDNNActForward {
...
-    else
-      this->data->set_data_handle(data.get_data_handle());
+              input_mem->get_primitive_desc(), input_mem->get_data_handle()));
+    } else {
+      this->data->set_data_handle(input_mem->get_data_handle());
+    }
 
-    CHECK(fwd_pd.dst_primitive_desc() == output.get_primitive_desc());
-    if (this->out == nullptr)
+    CHECK(fwd_pd.dst_primitive_desc() == out_mem->get_primitive_desc());
```
Comment (TaoLv) [inline src/operator/nn/mkldnn/mkldnn_act.cc]: Why check this? out_mem is just created in L116 with the dst_primitive_desc. Do you think mkldnn will fail in doing that?

File: src/operator/nn/mkldnn/mkldnn_act.cc (hunk 2 above)
Comment (rongzha1) [reply src/operator/nn/mkldnn/mkldnn_act.cc]: OK  leave data only

File: src/operator/nn/mkldnn/mkldnn_act.cc (hunk 1 above)
Comment (rongzha1) [reply src/operator/nn/mkldnn/mkldnn_act.cc]: OK move to constructed function;

File: src/operator/nn/mkldnn/mkldnn_act.cc (hunk 3 above)
Comment (rongzha1) [reply src/operator/nn/mkldnn/mkldnn_act.cc]: OK will remove it

**Review by rongzha1 — COMMENTED** [review]
will modify
